src/main/java/rx/Single.java

```diff
@@ -2254,7 +2254,15 @@ public final Completable toCompletable() {
      */
     public final Single<T> timeout(long timeout, TimeUnit timeUnit, Single<? extends T> other, Scheduler scheduler) {
         if (other == null) {
-            other = Single.<T> error(new TimeoutException());
+            // Use a defer instead of simply   other = Single.error(new TimeoutException())
```

Comment:
I'd inline this into `SingleTimeout`, that way, there are no extra instances unless necessary.